### DISCUSSION TITLE
Digital Ocean Streaming Error

### DISCUSSION THREAD

**Discussion by ThrudUK** [mastodon/mastodon#23003] [Q&A]
I've looked through the post here ; https://discourse.joinmastodon.org/t/not-getting-new-messages-unless-i-reload-the-page/3805/25

But not having much luck. 

I've run the DO Single click install *with* a separate database and ironed out most of the issues like admin not being able to see the administration setting when logged in and changing the setup to reference mastodon_production rather than defaultdb... but this one has me stumped!

I'm just seeing lots of this :

Firefox can’t establish a connection to the server at wss://fdom.social/api/v1/streaming/?.

Any help would be much appreciated!

**Comment by themullinator** [comment]
If you run `journalctl -u mastodon-streaming -n 25` on the server, what error messages are you getting?

**Comment by ThrudUK** [comment]
```
-- Logs begin at Sun 2023-01-08 20:35:28 UTC, end at Mon 2023-01-09 16:31:38 UTC. --
Jan 09 16:24:12 fdom node[4771]: WARN Starting worker 1
Jan 09 16:24:12 fdom node[4771]: WARN Worker 1 now listening on 127.0.0.1:4000
Jan 09 16:24:38 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:24:38 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:24:38 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:24:39 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:24:40 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:24:43 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:24:46 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:24:56 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:25:08 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:25:50 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:26:49 fdom node[4771]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:27:45 fdom systemd[1]: Stopping mastodon-streaming...
Jan 09 16:27:45 fdom node[4771]: WARN Worker 1 exiting
Jan 09 16:27:45 fdom node[4771]: WARN Worker 1 exiting
Jan 09 16:27:45 fdom systemd[1]: mastodon-streaming.service: Succeeded.
Jan 09 16:27:45 fdom systemd[1]: Stopped mastodon-streaming.
Jan 09 16:27:45 fdom systemd[1]: Started mastodon-streaming.
Jan 09 16:27:45 fdom node[4805]: WARN Starting streaming API server master with 1 workers
Jan 09 16:27:46 fdom node[4833]: WARN Starting worker 1
Jan 09 16:27:46 fdom node[4833]: WARN Worker 1 now listening on 127.0.0.1:4000
Jan 09 16:28:53 fdom node[4833]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:30:31 fdom node[4833]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off
Jan 09 16:30:52 fdom node[4833]: ERR! error: no pg_hba.conf entry for host "161.35.170.201", user "doadmin", database "mastodon_production", SSL off

```
Doesn't mean much to me!

**Comment by themullinator** [comment]
Okay, I finally managed to find time to sit down and solve this for my own deployment in DigitalOcean (I was getting pretty much the same pga_hba.conf errors.)

1. Ensure that your `.env.production` contains a value for `DB_SSLMODE`. While the Rails app appears to set a default value here, the Node server (used for streaming) does not. I set a value of `DB_SSLMODE=prefer` but I believe any valid SSL mode from https://www.postgresql.org/docs/10/libpq-ssl.html#LIBPQ-SSL-PROTECTION should work. 
2. Modify line 145 of `streaming/index.js`: https://github.com/mastodon/mastodon/blob/v4.0.2/streaming/index.js#L145. If you just want things to work and don't want to mess around with encrypting your connection, you can change this line to the following: `pgConfigs.production.ssl = { rejectUnauthorized: false };`. **I don't recommend this option if you're using a non-VPC connection to the DigitalOcean database.** 
3. If you _do_ want to encrypt your connection, download the DigitalOcean CA Certificate from your Managed Database dashboard, and then upload it to your Mastodon user directory. Make sure it's owned by `mastodon:mastodon`. Then edit line 145 of index.js with this instead: `pgConfigs.production.ssl = { ca: fs.readFileSync("relative/path/to/ca-certificate.crt") };` <img width="541" alt="DigitalOcean Dashboard" src="https://user-images.githubusercontent.com/1475149/215654516-4800e9d2-afed-4b1e-ad07-13d089c69475.png"> 
4. Restart the mastodon streaming service with `systemctl restart mastodon-streaming`.
